File: _includes/v2.1/app/insecure/basic-sample.js
var async = require('async');
var fs = require('fs');
var pg = require('pg');

// Connect to the "bank" database.
var config = {
    user: 'maxroach',
    host: 'localhost',
    database: 'bank',
    port: 26257
};

// Create a pool.
var pool = new pg.Pool(config);

pool.connect(function (err, client, done) {

    // Close communication with the database and exit.
    var finish = function () {
        done();
        process.exit();
    };

    if (err) {
        console.error('could not connect to cockroachdb', err);
        finish();
    }
    async.waterfall([
            function (next) {
                // Create the 'accounts' table.
                client.query('CREATE TABLE IF NOT EXISTS accounts (id INT PRIMARY KEY, balance INT);', next);
            },
            function (results, next) {
                // Insert two rows into the 'accounts' table.
                client.query('INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250);', next);
            },
            function (results, next) {
                // Print out account balances.
                client.query('SELECT id, balance FROM accounts;', next);
            },
        ],
        function (err, results) {
            if (err) {
                console.error('Error inserting into and selecting from accounts: ', err);
                finish();
            }

            console.log('Initial balances:');
            results.rows.forEach(function (row) {
                console.log(row);
            });

            finish();
        });
});

File: _includes/v1.1/app/basic-sample.rb
# Import the driver.
require 'pg'

# Connect to the "bank" database.
conn = PG.connect(user: 'maxroach', dbname: 'bank', host: 'localhost', port: 26257)

# Create the "accounts" table.
conn.exec('CREATE TABLE IF NOT EXISTS accounts (id INT PRIMARY KEY, balance INT)')

# Insert two rows into the "accounts" table.
conn.exec('INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250)')

# Print out the balances.
puts 'Initial balances:'
conn.exec('SELECT id, balance FROM accounts') do |res|
  res.each do |row|
    puts row
  end
end

# Close communication with the database.
conn.close()
